tests/test_dch.py

```diff
@@ -0,0 +1,79 @@
+import unittest
+
+import numpy as np
+
+from sklearn.datasets import load_diabetes
+from sklearn.decomposition import PCA
+
+from skcosmo.sample_selection import DirectionalConvexHull
+
+
+class TestDCH(unittest.TestCase):
+    def setUp(self):
+        self.X, self.y = load_diabetes(return_X_y=True)
+        self.T = PCA(n_components=4).fit_transform(self.X)
+        self.idx = [187, 57, 156, 123]
+        self.score_100 = [0.05926369, 0.03557203, 0.02328013]
+
+    def test_ground_truth(self):
+        """
+        Check that DCH selects correct vertices and gets correct scores.
+        """
+        selector = DirectionalConvexHull()
+        selector.fit(self.T, self.y)
+        self.assertTrue(np.allclose(selector.selected_idx_, self.idx))
+        scores = selector.score(self.T, self.y)
+        self.assertTrue(np.allclose(self.score_100, scores[100], atol=1e-4))
```

Comment:
`atol=1e-4` is very large given all the values are around 1e-2. Is this necessary? Or could we use something more stringent like `rtol=1e-6`?

Reply:
We put now rtol 1e-6

@victorprincipe had before larger differences than 1e-6 after working on it a month and probably updating the packages. We suspected that it is the scipy version, since the only thing that has an influence on the score function is the interpolator function `scipy.interpolate.griddata` (the selected idx are the same). So we did a small check (see at the end), but it did not seem that this is due to scipy. We do not know at the moment what the reason for the initial difference was. We just use the new reference values with rtol 1e-6 and investigate it further in case we can reproduce it

Checked different scipy versions
```python
import numpy as np
import scipy

from sklearn.datasets import load_diabetes
from sklearn.decomposition import PCA

from skcosmo.sample_selection import DirectionalConvexHull

X, y = load_diabetes(return_X_y=True)
T = PCA(n_components=4).fit_transform(X)
selector = DirectionalConvexHull()
selector.fit(T, y)
scores = selector.score(T, y)
X, y = load_diabetes(return_X_y=True)
T=PCA(n_components=4).fit_transform(X)
print("scipy version", scipy.__version__)
print("scores100", scores[100])
```

Outputs
```
scipy version 1.4.0
scores100 [0.05926369 0.03557203 0.02328013]

scipy version 1.6.0
scores100 [0.05926369 0.03557203 0.02328013]

scipy version 1.9.0
scores100 [0.05926369 0.03557203 0.02328013]

```

Reply:
okay the difference we saw seems to be because the diabetes dataset has changed a bit in version 1.1.

https://github.com/scikit-learn/scikit-learn/commit/379919dc591b487b56bbf8bd0dad5dd239e09753

I checked sklearn versions 1.0.2 and 1.2.0rc1 locally and get same difference.

This comment on the issue seems to agree with our observation.
https://github.com/scikit-learn/scikit-learn/pull/16605#issuecomment-894319647

I stop here and just update the new reference values for newest sklearn version

Reply:
That sounds good to me! Maybe also increase the minimal require sklearn version in the requirements.

Reply:
Super annoying, scikit-learn 1.1 is not supported anymore for python 3.7. Should we drop it or just use a different dataset for now and make this decision later?

Reply:
According to [NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html), Python 3.7 support has been dropped last year. So we could also decide to follow the rest of the ecosystem and drop it here as well.